app/components/RecentEvents.tsx

```diff
@@ -0,0 +1,204 @@
+import moment from 'moment';
+import { RecentEventsService, IRecentEvent } from 'services/recent-events';
+import TsxComponent from './tsx-component';
+import { Inject } from 'services/core';
+import { $t } from 'services/i18n';
+import styles from './RecentEvents.m.less';
+
+const getName = (event: IRecentEvent) => {
+  if (event.gifter) return event.gifter;
+  if (event.from) return event.from;
+  return event.name;
+};
+
+@Component({})
+export default class RecentEvents extends TsxComponent<{}> {
+  @Inject() recentEventsService: RecentEventsService;
+
+  queuePaused = false;
+
+  get recentEvents() {
+    console.log(this.recentEventsService.state.recentEvents);
+    return this.recentEventsService.state.recentEvents;
+  }
+
+  get muted() {
+    return this.recentEventsService.state.muted;
+  }
+
+  formatMoney(amount: string, type: string) {
+    const prefix = type === 'donation' ? '$' : '';
+    const numAmount = Number.parseFloat(amount);
+    return `${prefix}${type === 'donation' ? numAmount.toFixed(2) : numAmount.toFixed(0)}`;
+  }
+
+  eventString(event: IRecentEvent) {
+    return this.recentEventsService.getEventString(event);
+  }
+
+  repeatAlert(event: IRecentEvent) {
+    return this.recentEventsService.repeatAlert(event);
+  }
+
+  popoutRecentEvents() {
+    return this.recentEventsService.openRecentEventsWindow();
+  }
+
+  popoutMediaShare() {
+    return this.recentEventsService.openRecentEventsWindow(true);
+  }
+
+  muteEvents() {
+    return this.recentEventsService.toggleMuteEvents();
+  }
+
+  skipAlert() {
+    return this.recentEventsService.skipAlert();
+  }
+
+  async toggleQueue() {
+    try {
+      this.queuePaused
+        ? await this.recentEventsService.unpauseAlertQueue()
+        : await this.recentEventsService.pauseAlertQueue();
+      this.queuePaused = !this.queuePaused;
+    } catch (e) {}
+  }
+
+  render(h: Function) {
+    return (
+      <div class={styles.container}>
+        <Toolbar
+          popoutMediaShare={() => this.popoutMediaShare()}
+          popoutRecentEvents={() => this.popoutRecentEvents()}
+          muteEvents={() => this.muteEvents()}
+          skipAlert={() => this.skipAlert()}
+          toggleQueue={() => this.toggleQueue()}
+          queuePaused={this.queuePaused}
+          muted={this.muted}
+        />
+        <div class={styles.eventContainer}>
+          {this.recentEvents &&
+            this.recentEvents.map(event => (
+              <EventCell
+                event={event}
+                repeatAlert={this.repeatAlert.bind(this)}
+                eventString={this.eventString.bind(this)}
+              />
+            ))}
+        </div>
+      </div>
+    );
+  }
+}
+
+interface IToolbarProps {
+  popoutMediaShare: Function;
+  popoutRecentEvents: Function;
+  muteEvents: Function;
+  skipAlert: Function;
+  toggleQueue: Function;
+  queuePaused: boolean;
+  muted: boolean;
+}
+
+// TODO: Refactor into stateless functional component
```

Comment:
Honestly, I don't think it gives us any profit to split a single component into multiple components if you are not going to reuse this sub-components. Especially if you have only a single instance of sub-component like `Toolbar`. As a result, you need extra boilerplate code to transfer shared props between parent and child.

I'm not going to say you should refactor this now. But I don't think it should be refactored into `stateless functional component`. Our stateless components are in the `components/shared` folder, and they are really generic